core/block_validator.go

```diff
@@ -98,7 +124,45 @@ func (v *BlockValidator) ValidateState(block, parent *types.Block, statedb *stat
 	if root := statedb.IntermediateRoot(v.config.IsEIP158(header.Number)); header.Root != root {
 		return fmt.Errorf("invalid merkle root (remote: %x local: %x)", header.Root, root)
 	}
-	return nil
+
+	// Validate 3 roots from block committed on root chain against the received roots
+	// and throw an error if they don't match
+	validate := func() error {
+		tblock, err := v.rootchain.GetBlock(baseCallOpt, block.Difficulty(), block.Number())
```

Comment:
In case of `v.rootchain == nil`, skip the validation. It is useful when blockchain without RootChain contract is required.